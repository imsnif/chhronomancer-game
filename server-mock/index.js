import playerStore from '../stores/player'
import timelineStore from '../stores/timeline'
import powerStore from '../stores/power'
import statsStore from '../stores/stats'

const timelines = [
  {name: 'Timeline 1', type: 'steal', players: []},
  {name: 'Timeline 2', type: 'assist', players: []},
  {name: 'Timeline 3', type: 'prevent', players: []},
  {name: 'Timeline 4', type: 'reset', players: []}
]

const players = [
  {
    id: 1,
    name: 'Aram',
    items: [
      {name: 'assist', source: false},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 4'},
      {name: 'reset', source: 'Timeline 4'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: false},
      {name: 'unlock', source: false}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_me_s.png')
  },
  {
    id: 2,
    name: 'Tom',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 1'},
      {name: 'unlock', source: false}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_tom_s.png')
  },
  {
    id: 3,
    name: 'Gondollieri',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 3'},
      {name: 'unlock', source: 'Timeline 1'}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_gondollieri_s.png')
  },
  {
    id: 4,
    name: 'Simmons',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: false},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 2'},
      {name: 'unlock', source: 'Timeline 2'}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_simmons_s.png')
  }
]

const powers = [
  {
    playerId: 1,
    timelineName: 'Timeline 1',
    name: 'Resetting',
    startTime: Date.now(),
    endTime: Date.now() + 100000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 1, score: 8},
      {id: 2, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 9}
    ]
  },
  {
    playerId: 1,
    timelineName: 'Timeline 2',
    name: 'Locking',
    startTime: Date.now(),
    endTime: Date.now() + 400000,
    target: {
      type: 'timeline',
      name: 'Timeline 2'
    },
    allies: [
      {id: 1, score: 8},
      {id: 2, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 9}
    ]
  },
  {
    playerId: 2,
    timelineName: 'Timeline 1',
    name: 'Resetting',
    startTime: Date.now(),
    endTime: Date.now() + 1000000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 2, score: 1},
      {id: 1, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 2}
    ]
  },
  {
    playerId: 3,
    timelineName: 'Timeline 1',
    name: 'Combining',
    startTime: Date.now(),
    endTime: Date.now() + 2000000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 3, score: 1},
      {id: 1, score: 2}
    ],
    enemies: [
      {id: 2, score: 2},
      {id: 4, score: 5}
    ]
  },
  {
    playerId: 4,
    timelineName: 'Timeline 1',
    name: 'Locking',
    startTime: Date.now(),
    endTime: Date.now() + 100000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 4, score: 7},
      {id: 2, score: 2}
    ],
    enemies: [
      {id: 2, score: 2},
      {id: 1, score: 5}
    ]
  }
]

export default function mockServer () {
  timelines.forEach((t) => timelineStore.addTimeline(t))
  players.forEach(p => {
    playerStore.addPlayer(p)
  })
  statsStore.changePlayerId(1)
  timelineStore.addPlayer('Timeline 1', 1)
  timelineStore.addPlayer('Timeline 1', 2)
  timelineStore.addPlayer('Timeline 2', 2)
  timelineStore.addPlayer('Timeline 3', 2)
  timelineStore.addPlayer('Timeline 1', 3)
  timelineStore.addPlayer('Timeline 2', 3)
  timelineStore.addPlayer('Timeline 3', 3)
  timelineStore.addPlayer('Timeline 1', 4)
  timelineStore.addPlayer('Timeline 3', 4)
  powers.forEach(p => powerStore.addPower(p))
  timelineStore.changeTimelineLock('Timeline 1', true)
}
