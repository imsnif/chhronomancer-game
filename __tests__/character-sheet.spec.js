import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))
jest.useFakeTimers()

afterEach(() => {
  const clockStore = require('../stores/clock').default
  jest.useRealTimers()
  clockStore.destroy()
  jest.useFakeTimers()
})

test('Character-Sheet (default props) => renders correctly', () => {
  require('../server-mock')
  const CharacterSheet = require('../components/character-sheet').default
  const tree = renderer.create(
    <CharacterSheet />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Character-Sheet (no player) => renders empty', () => {
  require('../server-mock')
  const CharacterSheet = require('../components/character-sheet').default
  const statsStore = require('../stores/stats').default
  statsStore.changePlayerId('foo')
  const tree = renderer.create(
    <CharacterSheet />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
