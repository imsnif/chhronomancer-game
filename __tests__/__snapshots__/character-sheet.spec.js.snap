// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character-Sheet (default props) => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#141313",
      "flexDirection": "column",
      "height": "100%",
      "paddingBottom": 5,
      "paddingTop": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#329932",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 4,
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
        }
      }
      style={
        Object {
          "backgroundColor": "#141313",
          "borderColor": "black",
          "borderWidth": 1,
          "flex": 1,
          "height": "95%",
          "margin": 1,
          "resizeMode": "stretch",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "green",
            "height": "100%",
            "opacity": 0.5,
            "width": "100%",
          }
        }
      />
    </Image>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#329932",
          "color": "#141313",
          "flex": 5,
          "fontFamily": "telegrama_raw",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Aram
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#329932",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 4,
        "marginLeft": 15,
        "marginRight": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#329932",
          "color": "#141313",
          "flex": 1,
          "fontFamily": "telegrama_raw",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Actions:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#329932",
          "color": "#141313",
          "flex": 1,
          "fontFamily": "telegrama_raw",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      10
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 9,
        "flexDirection": "column",
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginBottom": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#329932",
            "borderColor": "#329932",
            "borderRadius": 0,
            "borderWidth": 0.5,
            "flex": 1,
            "flexDirection": "column",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#0e0e0e",
                "flex": 7,
                "fontFamily": "telegrama_raw",
                "paddingLeft": 5,
                "textAlign": "left",
              }
            }
          >
            Items
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#0e0e0e",
                "flex": 3,
                "fontFamily": "telegrama_raw",
                "textAlign": "center",
              }
            }
          >
            Cap: 5/7
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#141313",
              "borderColor": "#141313",
              "borderRadius": 0,
              "borderWidth": 0.5,
              "flex": 13,
              "flexDirection": "column",
              "margin": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginTop": 2,
                "minHeight": 27,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "height": undefined,
                  "marginLeft": 3,
                  "resizeMode": "stretch",
                  "width": undefined,
                }
              }
            />
            <View
              style={
                Object {
                  "flex": 12,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#00ff00",
                    "flex": 5,
                    "fontFamily": "telegrama_raw",
                    "paddingLeft": 9,
                    "paddingTop": 3,
                  }
                }
              >
                Steal
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "paddingTop": 5,
                  }
                }
              >
                ?
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginTop": 2,
                "minHeight": 27,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "height": undefined,
                  "marginLeft": 3,
                  "resizeMode": "stretch",
                  "width": undefined,
                }
              }
            />
            <View
              style={
                Object {
                  "flex": 12,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 5,
                    "fontFamily": "telegrama_raw",
                    "paddingLeft": 9,
                    "paddingTop": 3,
                  }
                }
              >
                Assist
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "paddingTop": 5,
                  }
                }
              >
                ?
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginTop": 2,
                "minHeight": 27,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "height": undefined,
                  "marginLeft": 3,
                  "resizeMode": "stretch",
                  "width": undefined,
                }
              }
            />
            <View
              style={
                Object {
                  "flex": 12,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#00ff00",
                    "flex": 5,
                    "fontFamily": "telegrama_raw",
                    "paddingLeft": 9,
                    "paddingTop": 3,
                  }
                }
              >
                Prevent (x2)
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "paddingTop": 5,
                  }
                }
              >
                ?
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginTop": 2,
                "minHeight": 27,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "height": undefined,
                  "marginLeft": 3,
                  "resizeMode": "stretch",
                  "width": undefined,
                }
              }
            />
            <View
              style={
                Object {
                  "flex": 12,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#00ff00",
                    "flex": 5,
                    "fontFamily": "telegrama_raw",
                    "paddingLeft": 9,
                    "paddingTop": 3,
                  }
                }
              >
                Reset (x2)
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "paddingTop": 5,
                  }
                }
              >
                ?
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginTop": 2,
                "minHeight": 27,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "height": undefined,
                  "marginLeft": 3,
                  "resizeMode": "stretch",
                  "width": undefined,
                }
              }
            />
            <View
              style={
                Object {
                  "flex": 12,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 5,
                    "fontFamily": "telegrama_raw",
                    "paddingLeft": 9,
                    "paddingTop": 3,
                  }
                }
              >
                Lock
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "paddingTop": 5,
                  }
                }
              >
                ?
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "marginTop": 2,
                "minHeight": 27,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "height": undefined,
                  "marginLeft": 3,
                  "resizeMode": "stretch",
                  "width": undefined,
                }
              }
            />
            <View
              style={
                Object {
                  "flex": 12,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 5,
                    "fontFamily": "telegrama_raw",
                    "paddingLeft": 9,
                    "paddingTop": 3,
                  }
                }
              >
                Unlock
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#329932",
                    "flex": 1,
                    "fontFamily": "telegrama_raw",
                    "paddingTop": 5,
                  }
                }
              >
                ?
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#329932",
            "borderColor": "#329932",
            "borderRadius": 0,
            "borderWidth": 0.5,
            "flex": 1,
            "flexDirection": "column",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#0e0e0e",
                "flex": 7,
                "fontFamily": "telegrama_raw",
                "textAlign": "center",
              }
            }
          >
            Iterations
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#141313",
              "borderColor": "#141313",
              "borderRadius": 0,
              "borderWidth": 0.5,
              "flex": 13,
              "flexDirection": "column",
              "margin": 1,
            }
          }
        >
          <RCTScrollView>
            <View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "flexDirection": "column",
                      "height": 25,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#00ff00",
                        "flex": 1,
                        "fontFamily": "telegrama_raw",
                        "fontSize": 8,
                        "marginTop": 5,
                        "paddingRight": 9,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    Timeline 1
                  </Text>
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#329932",
                            "flex": 1,
                            "fontFamily": "telegrama_raw",
                            "fontSize": 6,
                            "marginLeft": 9,
                            "paddingTop": 2,
                            "textAlign": "left",
                          }
                        }
                      >
                        Resetting
                      </Text>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#329932",
                            "flex": 7,
                            "fontFamily": "telegrama_raw",
                            "fontSize": 6,
                            "paddingTop": 2,
                            "textAlign": "left",
                          }
                        }
                      >
                        | |==------------------------------------------------|   1% 00:01:39
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "flexDirection": "column",
                      "height": 25,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#00ff00",
                        "flex": 1,
                        "fontFamily": "telegrama_raw",
                        "fontSize": 8,
                        "marginTop": 5,
                        "paddingRight": 9,
                        "paddingTop": 2,
                        "textAlign": "center",
                      }
                    }
                  >
                    Timeline 2
                  </Text>
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#329932",
                            "flex": 1,
                            "fontFamily": "telegrama_raw",
                            "fontSize": 6,
                            "marginLeft": 9,
                            "paddingTop": 2,
                            "textAlign": "left",
                          }
                        }
                      >
                        Locking
                      </Text>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#329932",
                            "flex": 7,
                            "fontFamily": "telegrama_raw",
                            "fontSize": 6,
                            "paddingTop": 2,
                            "textAlign": "left",
                          }
                        }
                      >
                        | |==------------------------------------------------|   1% 00:06:39
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Character-Sheet (no player) => renders empty 1`] = `null`;
